packages/fxa-auth-server/scripts/move-customers-to-new-plan-v2/move-customers-to-new-plan-v2.ts

```diff
@@ -172,16 +198,38 @@ export class CustomerPlanMover {
       throw new Error("Customer is pending renewal right now!");
     }
 
-    const billedAt = new Date(invoice.created * 1000);
     const oneDayMs = 1000 * 60 * 60 * 24;
-    const timeSinceBillMs = new Date().getTime() - billedAt.getTime();
+
+    const lastBilledAt = new Date(subscription.current_period_start * 1000);
+    const timeSinceBillMs = new Date().getTime() - lastBilledAt.getTime();
     const daysSinceBill = Math.floor(timeSinceBillMs / oneDayMs);
-    const refundAmount = daysSinceBill * this.proratedRefundRate;
+
+    const nextBillAt = new Date(subscription.current_period_end * 1000);
+    const timeUntilBillMs = nextBillAt.getTime() - new Date().getTime();
+    const daysUntilBill = Math.floor(timeUntilBillMs / oneDayMs);
+
+    const refundAmount = daysUntilBill * this.proratedRefundRate;
+
+    return { refundAmount, daysUntilBill, daysSinceBill, invoice };
+  }
+
+  /**
+   * Attempt to refund customer for latest bill considering the prorated refund amount
+   * @param subscription The subscription to cancel
+   */
+  async attemptRefund(subscription: Stripe.Subscription) {
+    const calculation = await this.calculateRefundAmount(subscription);
+    const refundAmount = calculation.refundAmount;
+    const invoice = calculation.invoice;
 
     if (refundAmount > invoice.amount_paid) {
       throw new Error(`Will not refund ${invoice.id} for ${refundAmount} as it would eclipse the amount paid on the invoice`);
     }
 
+    if (refundAmount <= 0) {
```

Comment:
I believe Stripe's general minimum transaction amount is $0.50. Its possible we'll run into the edge case of failed refunds for folks who have nonzero-but-small credit amounts after cancellation

Reply:
(Unfortunately I don't think it's exactly the same for all currencies)